app/controllers/redirections_controller.rb

```diff
@@ -73,33 +72,33 @@ def tagged_log(message)
   end
 
   def ensure_referrer_is_not_localhost
-    if HostValidator.new(referrer).invalid?
-      redirect_to page_path("localhost")
+    if creating_new_redirection?
```

Comment:
If you like you could also put these in the `before_action`s:

```rb
before_action :ensure_referrer_is_not_localhost, if :creating_new_redirection?
```

Reply:
Yeah, but:

1. I dislike the DSL for stuff that's already in Ruby
2. I want to move these to their own classes someday and it's a bit easier if everything is in the function already.